### PR TITLE
Another version of FMVP imply.

### PR DESCRIPTION
**Pull request by oleksii-demedetskyi**
This is simplified vision of FVMP.

Note, that there is no Disposable support, 
but general code layout should be clear to you.

### CODE REVIEW

File: test/TextUpdater.swift [hunk 1]
```diff
@@ -1,27 +1,48 @@
...
+    typealias Services
+    typealias Presenters
     
-    struct ViewModelPresenter {
+    static func ViewModel(services: Services, _ presenters: Presenters);
+}
+
+extension ViewModel {
+    typealias UnpresentedViewModel = Presenters -> ()
+    static func ViewModel(services: Services)(presenters: Presenters) {
```
Comment (xNekOIx) [inline test/TextUpdater.swift:12]: you said currying will not be avaialble soon

File: test/TextUpdater.swift [hunk 2]
```diff
@@ -1,27 +1,48 @@
...
+}
+
+class TextUpdaterViewController: UIViewController {
+    @IBOutlet weak var label: UILabel!
+    @IBOutlet weak var textField: UITextField!
+    
+    var viewModel: TextUpdater.UnpresentedViewModel?
+    
+    override func viewDidLoad() {
+        viewModel?(TextUpdater.Presenters(
```
Comment (xNekOIx) [inline test/TextUpdater.swift:45]: ``` super.viewDidLoad()```
why optional?
```

**Comment by xNekOIx** [comment]
cool :+1:

File: test/TextUpdater.swift (hunk 1 above)
Comment (oleksii-demedetskyi) [inline test/TextUpdater.swift:12]: Just an illustration for `ViewModel as a function`

File: test/TextUpdater.swift (hunk 2 above)
Comment (oleksii-demedetskyi) [inline test/TextUpdater.swift:45]: It is not presented during whole VC lifecycle.

**Comment by oleksii-demedetskyi** [comment]
I am going add some networking code here. Just for illustration.

File: test/TextUpdater.swift (hunk 2 above)
Comment (xNekOIx) [inline test/TextUpdater.swift:45]: I meant why not implicitly unwrapped call
so you will know if you forgot to set presenters

File: test/APIService.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+//  test
+//
+//  Created by Алексей Демедецкий on 23.01.16.
+//  Copyright © 2016 Silicon Valley Insight. All rights reserved.
+//
+
+import Foundation
+
+enum APIService {
+    static func requestDefinitions(ofWord word: String, callback: String? -> ()) {
```
Comment (xNekOIx) [inline test/APIService.swift:12]: mb BrightFuture?

File: test/TextUpdater.swift (hunk 2 above)
Comment (oleksii-demedetskyi) [inline test/TextUpdater.swift:45]: what else options do we have?

File: test/APIService.swift (hunk 3 above)
Comment (oleksii-demedetskyi) [inline test/APIService.swift:12]: I want something extremely small...

**Comment by xNekOIx** [comment]
@AlexeyDemedetskiy have you done already? can I merge this?

File: test/TextUpdater.swift (hunk 2 above)
Comment (xNekOIx) [inline test/TextUpdater.swift:45]:
```
viewModel!(
```

as it was in previous version :)

**Comment by oleksii-demedetskyi** [comment]
sure. I will continue my work. I have plans for adding more screens.

**Pull request closed without merging**